File: quiz/question.py
from enum import Enum
from typing import Optional
from dataclasses import dataclass


class QuestionState(Enum):
    WAIT = 0
    ASKED = 1
    LOCKED = 2
    REVEALED = 3


class QuestionKind(Enum):
    TEXT = 301
    IMAGE = 302
    AUDIO = 303
    VIDEO = 304


@dataclass
class Question:
    number: int
    state: QuestionState
    kind: QuestionKind
    text: str
    answer: str
    filename: Optional[str]
    mimetype: Optional[str]

File: quiz/static/question.js
"use strict";

function Question(number, kind, text, media, state, guess, wip, answer) {
    this.number = number;
    this.kind = kind;
    this.text = text;
    this.media = media;
    this.state = state;
    this.guess = guess;
    this.wip = wip;
    this.answer = answer;
    Object.seal(this);
}

Question.create = function (number, kind, text, media) {
    return new Question(
        number,
        kind,
        text,
        media,
        QUESTION_STATE_OPEN,
        "",
        null,
        null,
    );
};

Question.prototype.toJSON = function () {
    return {
        "number": this.number,
        "kind": this.kind,
        "text": this.text,
        "media": this.media,
        "state": this.state,
        "guess": this.guess,
        "wip": this.wip,
        "answer": this.answer,
    };
};
